tools/ai-evals/azsdk-mcp/Evaluators/ExpectedToolInputEvaluator.cs

```diff
@@ -0,0 +1,173 @@
+using System.Text.Json;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.AI.Evaluation;
+using Azure.Sdk.Tools.McpEvals.Models;
+
+namespace Azure.Sdk.Tools.McpEvals.Evaluators
+{
+   public class ExpectedToolInputEvaluator : IEvaluator
+   {
+        public const string ExpectedToolInputMetricName = "Azure SDK MCP";
+        public IReadOnlyCollection<string> EvaluationMetricNames => [ExpectedToolInputMetricName];
+
+        public ValueTask<EvaluationResult> EvaluateAsync(
+            IEnumerable<ChatMessage> messages,
+            ChatResponse modelResponse,
+            ChatConfiguration? chatConfiguration = null,
+            IEnumerable<EvaluationContext>? additionalContext = null,
+            CancellationToken cancellationToken = default)
+        {
+            // Create the metric
+            var metric = new BooleanMetric(ExpectedToolInputMetricName);
+            var result = new ValueTask<EvaluationResult>(new EvaluationResult(metric));
+
+            if (additionalContext?.OfType<ExpectedToolInputEvaluatorContext>().FirstOrDefault()
+                is not ExpectedToolInputEvaluatorContext context)
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"A value of type {nameof(ExpectedToolInputEvaluatorContext)} was not found in the {nameof(additionalContext)} collection."));
+
+                return result;
+            }
+
+            // Get tool calls to compare them
+            var expectedToolCalls = GetToolContent(context.ChatMessages);
+            var actualToolCalls = GetToolContent(modelResponse.Messages);
+
+            // Make sure we have tool calls to compare
+            if (!expectedToolCalls.Any())
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
+                        $"No Tool calls detected inside of {nameof(additionalContext)} collection."));
+
+                return result;
+            }
+
+            if (!actualToolCalls.Any())
+            {
+                metric.AddDiagnostics(
+                    EvaluationDiagnostic.Error(
```

Comment:
Can you include information about expected Tool calls in this error message so it's very clear that actual tool call is empty when the expectation is a number of tool calls.